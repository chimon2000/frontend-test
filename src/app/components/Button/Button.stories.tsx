import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { action } from "@storybook/addon-actions";
import { withKnobs, boolean, text } from "@storybook/addon-knobs";

import { Button } from "./Button";

const stories = storiesOf("Components", module);
stories.addDecorator(withKnobs);
stories.add(
  "Button",
  withInfo({ inline: true })(() => (
    <Button
      secondary={boolean("Secondary", false)}
      disabled={boolean("Disabled", false)}
      block={boolean("Block", false)}
      onClick={action("click!")}
      className={text("Class Name", null)}
    >
      {text("Button Text", "Button Text")}
    </Button>
  ))
);
