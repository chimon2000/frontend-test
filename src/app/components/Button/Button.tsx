import * as React from "react";

import { joinClassList } from "../../helpers";

export interface ButtonProps {
  onClick: (e: React.SyntheticEvent) => void;
  secondary?: boolean;
  disabled?: boolean;
  block?: boolean;
  children: React.ReactNode;
  className?: string;
  id?: string;
}

export const Button: React.FunctionComponent<ButtonProps> = ({
  onClick,
  secondary,
  disabled,
  block,
  children,
  className
}) => {
  const classes = [
    `button`,
    `${secondary ? "button--secondary" : ""}`,
    `${className || ""}`
  ];
  return (
    <button
      disabled={disabled}
      className={joinClassList(classes)}
      onClick={onClick}
    >
      {children}
    </button>
  );
};
