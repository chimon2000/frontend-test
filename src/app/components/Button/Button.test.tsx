import * as React from "react";
import { shallow } from "enzyme";

import { Button } from "./Button";

describe("<Button />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(<Button onClick={() => {}}>Content</Button>);

    expect(wrapper).toMatchSnapshot();
  });

  it("Calls onClick method", () => {
    let triggered = false;
    const clickMethod = () => {
      triggered = true;
    };
    const wrapper = shallow(<Button onClick={clickMethod}>Content</Button>);
    const button = wrapper.find("button").at(0);
    button.simulate("click");
    expect(triggered).toEqual(true);
  });

  it("Disables with disabled flag", () => {
    const wrapper = shallow(
      <Button disabled onClick={() => {}}>
        Content
      </Button>
    );

    expect(wrapper.prop("disabled")).toBe(true);
  });
});
