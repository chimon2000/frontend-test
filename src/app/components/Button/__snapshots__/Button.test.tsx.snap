// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> doesn't blow up 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Content
</button>
`;
